cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,495 @@
+    }
+    if (sort) {
+      expect(savedObjectAttributes.sort).eqls(APPLIED_SORT);
+    }
+    expect(searchSource).match(
+      // all special characters must be escaped
+      new RegExp(`"query":"${queryString.replaceAll(/([*{}])/g, (char) => `\\${char}`)}"`)
+    );
+    expect(searchSource).match(new RegExp(`"language":"${apiLanguage}"`));
+    expect(searchSource).match(new RegExp(`"title":"${dataset.replace('*', '\\*')}"`));
+    expect(searchSource).match(new RegExp(`"type":"${datasetType}"`));
+
+    if (histogram) {
+      expect(searchSource).match(/"calendar_interval":"1w"/);
+    }
+    if (filters) {
+      expect(searchSource).match(
+        new RegExp(`"match_phrase":\{"${APPLIED_FILTERS.field}":"${APPLIED_FILTERS.value}"\}`)
+      );
+    }
+  });
+};
+
+export const runSavedSearchTests = () => {
+  describe('saved search', () => {
+    beforeEach(() => {
+      // Load test data
+      cy.setupTestData(
+        SECONDARY_ENGINE.url,
+        [
+          `cypress/fixtures/query_enhancements/data-logs-1/${INDEX_WITH_TIME_1}.mapping.json`,
+          `cypress/fixtures/query_enhancements/data-logs-2/${INDEX_WITH_TIME_2}.mapping.json`,
+        ],
+        [
+          `cypress/fixtures/query_enhancements/data-logs-1/${INDEX_WITH_TIME_1}.data.ndjson`,
+          `cypress/fixtures/query_enhancements/data-logs-2/${INDEX_WITH_TIME_2}.data.ndjson`,
+        ]
+      );
+      // Add data source
+      cy.addDataSource({
+        name: datasourceName,
+        url: SECONDARY_ENGINE.url,
+        authType: 'no_auth',
+      });
+      // Grab the data source ID
+      cy.contains('a', datasourceName).click();
+      cy.url().then((url) => {
+        const urlWithoutSearchParams = url.split('?')[0];
+        // split the URL into parts and filter out the empty ones
+        const urlParts = urlWithoutSearchParams.split('/').filter((parts) => !!parts.length);
+        datasourceId = urlParts[urlParts.length - 1];
+      });
+
+      // Create workspace
+      cy.deleteWorkspaceByName(workspaceName);
+      cy.visit('/app/home');
+      cy.createInitialWorkspaceWithDataSource(datasourceName, workspaceName);
+      cy.createWorkspaceIndexPatterns({
+        workspaceName: workspaceName,
+        indexPattern: INDEX_PATTERN_WITH_TIME.replace('*', ''),
+        timefieldName: 'timestamp',
+        dataSource: datasourceName,
+        isEnhancement: true,
+      });
+      cy.url().then((url) => {
+        const urlWithoutSearchParams = url.split('?')[0];
+        // split the URL into parts and filter out the empty ones
+        const urlParts = urlWithoutSearchParams.split('/').filter((parts) => !!parts.length);
+        // the index pattern path has a # at the end, so stripping it
+        indexPatternId = urlParts[urlParts.length - 1].replace('#', '');
+
+        const workspaceIdIndex = urlParts.findIndex((part) => part === 'w') + 1;
+        workspaceId = urlParts[workspaceIdIndex];
```

Comment:
we can grab all three from discover page. make a util function out of it

Reply:
hmm making the util is proving to be more tedius then i originally thought:

```
export const parseDataFromDiscoverPageUrl = (workspaceName, isEnhancement = false) => {
  cy.navigateToWorkSpaceSpecificPage({
    workspaceName,
    page: 'discover',
    isEnhancement,
  });
  cy.url().then((url) => {
    const workspaceId = url.match(/\/w\/(.+)\/app/)[1];
    const dataSourceId = url.match(new RegExp(`dataSource:\\(id:'(${UUID_REGEX_STRING})'`))[1];
    const datasetId = url.match(new RegExp(`dataset:\\(.+\\),id:'(${UUID_REGEX_STRING})'`))[1];

    // what to do here inside this async function?
  });
};
```

Since we are within `cy.url().then(...)` async function, you cannot return the three ids from the outer function. There probably is a way to do it but it will require some deep dive.

But furthermore, those regex I am using is pretty gnarly and not sure whether it works for all use cases.

I think I will hold off on making a utility out of this for now unless someone has an idea

Reply:
We discussed offline, I will be leveraging aliases